articles/quickstart/backend/java-spring-security5/01-authorization.md

```diff
@@ -0,0 +1,317 @@
+---
+title: Authorization
+name: How to secure your API using Spring Security 5 and Auth0
+description: This tutorial demonstrates how to add authorization to a an API using Spring Security 5.
+budicon: 500
+topics:
+    - quickstart
+    - backend
+    - java
+    - spring-security
+github:
+    path: 01-Authorization-MVC
+contentType: tutorial
+useCase: quickstart
+---
+
+<%= include('../../../_includes/_api_auth_intro') %>
+
+<%= include('../_includes/_api_create_new') %>
+
+<%= include('../_includes/_api_auth_preamble') %>
+
+## Configure the Sample Project
+
+::: note
+This Quickstart uses Spring MVC. If you are using Spring WebFlux, the steps to secure an API are similar, but some of the implementation details are different. Refer to the [Spring Security WebFlux Sample Code](https://github.com/auth0-samples/auth0-spring-security5-api-sample/tree/master/01-Authorization-WebFlux) to see how integrate Auth0 with your Spring WebFlux API.
```

Comment:
Normally linking to a GH repo file/folder gets outdated rather quickly. If the changes are only on a single class and its annotations (as far as I remember from our conversation). Why not have a separate section, maybe at the end of this document, explaining that snippet?

Small typo 

```suggestion
This Quickstart uses Spring MVC. If you are using Spring WebFlux, the steps to secure an API are similar, but some of the implementation details are different. Refer to the [Spring Security WebFlux Sample Code](https://github.com/auth0-samples/auth0-spring-security5-api-sample/tree/master/01-Authorization-WebFlux) to see how to integrate Auth0 with your Spring WebFlux API.
```

Reply:
Regarding having a snippet for WebFlux instead of linking to a GH repo, the issue is Spring MVC and WebFlux are entirely different ways of running an application. So the code changes for this sample are relatively minor - the most notable difference is in the `SecurityConfig`, but there are other changes such as dependencies (use `webflux` dependencies), and APIController (return `Mono` or `Flux`, etc.). So if we have a snippet or another section for Spring WebFlux, it would also require updates to other snippets/sections of the article so people don't get the two mixed up.

Another alternative would be to separate into multiple parts - `Authorization (MVC)` and `Authorization (WebFlux)`, and providing separate documents for each. WDYT?

Reply:
> Another alternative would be to separate into multiple parts - Authorization (MVC) and Authorization (WebFlux), and providing separate documents for each. WDYT?

For context, the reason I did not do this to begin with is that doing so could make the flow of the Quickstart less natural, in that our Quickstarts typically build on each step (e.g., 1) Authorization, 2) Using your API, 3) Troubleshooting). But in this case, MVC and WebFlux are different; one is not a prerequisite for another.